src/components/ReportTransaction.js

```diff
@@ -76,14 +90,30 @@ class ReportTransaction extends Component {
     }
 
     render() {
+        const {
+            sessionEmail, action, currentUserName, participantName,
+        } = this.props;
+        const {
+            type: messageType, amount, currency, comment, paymentType,
+        } = action.originalMessage;
+        const messageText = this.props.translate(`iou.transactions.${messageType}`, {
+            comment,
+            amount: !_.isNaN(Number(amount)) ? this.props.numberFormat(
+                Math.abs(amount) / 100,
+                {style: 'currency', currency},
+            ) : '',
+            participant: sessionEmail === action.actorEmail ? participantName : currentUserName,
+            paymentType,
+        });
```

Comment:
I'll take a look and see if I can move it to a separate function? Do we have any examples where we're pulling data from Onyx without any component?

Reply:
Yes, but we could also create a utility method that takes the props from the component and returns the message maybe? 

I think the problem with doing it someplace else is that `withLocalize` also tells the component it needs to re-render.

Not sure if there is a clear way. Maybe think about it and let me know what you plan to do. If we can't come to any agreement we can just leave things as they are?

Reply:
Yeah. I did think of moving the whole message logic but couldn't come up with something that ensures translations work. 

Also if you notice, I am using `iouReport` in `IOUTransactions`, whereas `chatReport` in `IOUAction`. I do that because generally for group chats you'll have multiple participants whereas IOUReport has only 1:1 transactions.

Hence I had to settle for a copy-paste approach.

Reply:
>Also if you notice, I am using iouReport in IOUTransactions, whereas chatReport in IOUAction. I do that because generally for group chats you'll have multiple participants whereas IOUReport has only 1:1 transactions.

Not sure I follow. Do these reports have the exact same shape or not?

Reply:
In group chats what happens is, if I split 100$ with 5 people:
1. I'll see one message in group chat `Split 100$ with A, B, C, and D by E`
2. 1 transaction message per 1:1 chat between E and each of A to D is added.
3. `IOUReport` then has an option to click and see this transaction in `IOUModal`

So to extract participant info I pick the data from IOUReport in case of `IOUModal` where is it 1:1 only.

whereas in `IOUQuote`, I extract the first name from an array of participants based on `chatReport` object

Reply:
Hmm sorry, I'm pretty confused by this explanation and also don't see where the "report" is being referenced. 

One idea to make this more abstract is to make a `IOUMessageText` component then pass the necessary props to it. It can be used in both places. Can that work?

Reply:
Apologies for the back and forth. Yeah, that can work.  Will send an update by tomorrow.

Reply:
No worries at all. Thanks for your patience!